### PR TITLE
feat: new swiper for wui v5

### PR DESCRIPTION
**Pull request by jeromeraffin**
Related issue: https://github.com/WTTJ/welcome-ui/issues/1899

### CODE REVIEW

File: docs/pages/components/swiper.mdx [hunk 1]
```diff
@@ -19,238 +19,120 @@ import {
...
 
-## With custom pagination items
+## Without arrow buttons
 
-By providing the optional `renderPaginationItem` you can customise the defaults pagination bullets and render your own component. This callback takes a `props` param which is an object containning:
-
-- `idx`: the current pagination item
-- `pageIdx`: the currently selected slide (which you can use to show the active state for your pagination items)
-- `onClick`: callback to pass to your custom bullet
+By providing the optional `withArrows` with `[false, false]`, you can hide the navigation arrows on mobile/tablet and desktop.
```
Comment (cnairiwttj) [inline docs/pages/components/swiper.mdx:118]: I'm wondering for all these props if it wouldn't be clearer to have an object?
Like this:
```
withArrows = { mobile: false, desktop: false }
```
In my opinion this is more explicit but I don't know if it's more complex

File: packages/Swiper/src/index.tsx [hunk 2]
```diff
@@ -1,175 +1,168 @@
...
-        >
-          {nextButton}
-        </S.Next>
-      )}
-    </S.Wrapper>
-  )
-})
+  const firstPageToShow = centeredSlides
+    ? Math.floor(slidesLength / currentSlidesPerView / 2)
+    : Math.ceil(firstSlideToShow / currentSlidesPerView) - 1
```
Comment (cnairiwttj) [inline packages/Swiper/src/index.tsx:81]: Could you please just explain to me what this calculation is doing?
I don't understand why in the first case we do not take into account the firstSlideToShow?

Comment (theo-mesnil) [reply packages/Swiper/src/index.tsx]: @jeromeraffin please add some description :)

File: docs/pages/components/swiper.mdx (hunk 1 above)
Comment (jeromeraffin) [reply docs/pages/components/swiper.mdx:118]: Yep I agree, I changed all the props with an array for an object, thanks!

**Review by theo-mesnil — APPROVED** [review]
🔥

- Please add on upgrade.mdx page the breaking changed
- Add more description on complex function

File: packages/Swiper/src/index.tsx [hunk 3]
```diff
@@ -1,175 +1,181 @@
...
-        >
-          {nextButton}
-        </S.Next>
-      )}
-    </S.Wrapper>
-  )
-})
+  const firstPageToShow = centeredSlides
+    ? // if centeredSlides is true, we calculate which number is the middle page
+      Math.ceil(numberOfPage / 2)
```
Comment (cnairiwttj) [inline packages/Swiper/src/index.tsx:92]: Thanks for the explanation, it's much clearer!
Maybe we could say it in the doc that the firstSlideToShow and centeredSlides props cannot be combined?
I don't know if you've already written it

Comment (jeromeraffin) [reply packages/Swiper/src/index.tsx:92]: Yes you're right, I just add the info on the doc that `firstSlideToShow` won't be used if centeredSlides is true

File: packages/Swiper/src/index.tsx (hunk 2 above)
Comment (cnairiwttj) [reply packages/Swiper/src/index.tsx]: 🔥

File: docs/pages/components/swiper.mdx (hunk 1 above)
Comment (cnairiwttj) [reply docs/pages/components/swiper.mdx]: 🔥  thank you!

File: packages/Swiper/src/index.tsx (hunk 3 above)
Comment (cnairiwttj) [reply packages/Swiper/src/index.tsx:93]: That's perfect, thanks 🔥

**Review by cnairiwttj — APPROVED** [review]
🔥 Congrats for this huge work!

**Pull request merged**